financial-templates-lib/test/price-feed/CreatePriceFeed.js

```diff
@@ -0,0 +1,219 @@
+const { createPriceFeed } = require("../../price-feed/CreatePriceFeed");
+const { CryptoWatchPriceFeed } = require("../../price-feed/CryptoWatchPriceFeed");
+const { UniswapPriceFeed } = require("../../price-feed/UniswapPriceFeed");
+const { MedianizerPriceFeed } = require("../../price-feed/MedianizerPriceFeed");
+const { NetworkerMock } = require("./NetworkerMock");
+const winston = require("winston");
+
+contract("CreatePriceFeed.js", function(accounts) {
+  const { toChecksumAddress, randomHex } = web3.utils;
+
+  let mockTime = 1588376548;
+  let networker;
+
+  const apiKey = "test-api-key";
+  const exchange = "test-exchange";
+  const pair = "test-pair";
+  const lookback = 120;
+  const getTime = () => mockTime;
+  const minTimeBetweenUpdates = 60;
+  const twapLength = 180;
+  const uniswapAddress = toChecksumAddress(randomHex(20));
+
+  beforeEach(async function() {
+    networker = new NetworkerMock();
+    logger = winston.createLogger({
+      level: "info",
+      transports: []
+    });
+  });
+
+  it("No type", async function() {
+    const config = {
+      apiKey,
+      exchange,
+      pair,
+      lookback,
+      minTimeBetweenUpdates
+    };
+
+    assert.equal(await createPriceFeed(web3, logger, networker, getTime, config), null);
+  });
+
+  it("Valid CryptoWatch config", async function() {
+    const config = {
+      type: "cryptowatch",
```

Comment:
This is awesome. I really like how this has been generalized and the type can be specified.